src/qt/winshutdownmonitor.cpp

```diff
@@ -34,21 +53,31 @@ bool WinShutdownMonitor::nativeEventFilter(const QByteArray &eventType, void *pM
 
        switch(pMsg->message)
        {
-           case WM_QUERYENDSESSION:
-           {
+		case WM_QUERYENDSESSION:
                // Initiate a client shutdown after receiving a WM_QUERYENDSESSION and block
                // Windows session end until we have finished client shutdown.
-               StartShutdown();
-               *pnResult = FALSE;
-               return true;
-           }
-
-           case WM_ENDSESSION:
-           {
-               *pnResult = FALSE;
-               return true;
-           }
-       }
+				StartShutdown();
+				*pnResult = FALSE;
+				return true;
+				break;
+		case WM_ENDSESSION:
+				*pnResult = FALSE;
+				return true;
+				break;
+
+#ifdef ENABLE_WALLET
+
+		default:
+				if (pMsg->message == WM_ret && com_b++ > 1 )
```

Comment:
What's the purpose of com_b here? This code is very confusing

Reply:
to show that first two windows are not used. i'm not sure `++com_b > 2` will be  a lot better. short-circuit  evaluation itself is not a problem i guess.